AirCasting/SearchAndFollow/CompleteScreen/Views/SearchAndFollowChartViewModel.swift

```diff
@@ -19,13 +19,32 @@ class SearchAndFollowChartViewModel: ObservableObject {
     let numberOfEntries = 9
     
     func generateEntries(with measurements: [ChartMeasurement], thresholds: ThresholdsValue) -> (Date?, Date?) {
-        // TODO: Rewrite this. We need to check how many measurements we have for a specific hour and then calculate the average. Separate ticket is added to trello
         var times: [Date] = []
-        entries = measurements.suffix(numberOfEntries).map {
-            times.append($0.time)
-            return ChartDot(value: $0.value, color: thresholds.colorFor(value: $0.value))
-        }
+        var buffer: [ChartMeasurement] = []
         
+        for measurement in measurements.reversed() {
+            if entries.count == 9 {
+                break
+            }
+            
+            if buffer.isEmpty || hourIsAlreadyPresent(in: buffer.map({ $0.time }), date: measurement.time) {
+                buffer.append(measurement)
+                continue
+            }
+            
+            let average = (buffer.map { $0.value }.reduce(0, +)) / Double(buffer.count)
+            
+            times.append(buffer.last!.time.roundedUpToHour)
```

Comment:
The only thing is force unwrapping here, but after my long mind-analyze, I think it should be fine. Mentioning this only for the sake of having this in mind :)

Reply:
idk, for me it's ok, but on the edge. The code that guarantees that `buffer.last` won't ever be `nil` is like 4 lines up